### PR TITLE
STITCH-3383, STITCH-3632

### PR DESCRIPTION
**Pull request by jsflax**
- Add custom function auth provider
- Add function for user pass reset

### CODE REVIEW

**Comment by mburdette** [comment]
no PR title? lol

File: Core/StitchCoreAdminClient/Sources/StitchCoreAdminClient/AuthProviders/ProviderConfig.swift [hunk 1]
```diff
@@ -17,6 +17,13 @@ private enum UserpassCodingKeys: String, CodingKey {
     case resetPasswordURL = "resetPasswordUrl"
     case confirmEmailSubject
     case resetPasswordSubject
+    case runResetFunction
+    case resetFunctionId
+    case resetFunctionName
```
Comment (mburdette) [inline Core/StitchCoreAdminClient/Sources/StitchCoreAdminClient/AuthProviders/ProviderConfig.swift]: is this PR also supposed to have reset function stuff in it? seems like this should be in a different ticket. If this is a driveby, shouldn't it also contain confirmation function stuff as well?

**Comment by jsflax** [comment]
It combines both tickets– I'll add a description
